### PR TITLE
support png format images for object detection

### PR DESCRIPTION
**Pull request by dding3**
Fix https://github.com/intel-analytics/zoo/issues/301

### CODE REVIEW

**Comment by changlinzhang** [comment]
I think we'd better to replace `analytics-zoo/zoo/src/test/resources/png/4-1.PNG` with another image or user specific path, because this image is from our customer.

**Comment by jason-dai** [comment]
Any update?

File: zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ZooOpenCVMat.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+package com.intel.analytics.zoo.feature.image
+
+import com.intel.analytics.bigdl.opencv.OpenCV
+import com.intel.analytics.bigdl.transform.vision.image.opencv.OpenCVMat
+
+import org.opencv.core.{Mat, MatOfByte}
+import org.opencv.imgcodecs.Imgcodecs
+
+private[zoo] object ZooOpenCVMat {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/feature/image/ZooOpenCVMat.scala]: Change to `OpenCV` or sth like `OpenCVMethod`

**Comment by jason-dai** [comment]
Please also update the doc

File: pyzoo/zoo/feature/image/imageset.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def is_distributed(self):
 
     @classmethod
     def read(cls, path, sc=None, min_partitions=1, resize_height=-1,
-             resize_width=-1, bigdl_type="float"):
+             resize_width=-1, flags=-1, bigdl_type="float"):
```
Comment (jason-dai) [inline pyzoo/zoo/feature/image/imageset.py]: change the name to `image_codec`?

**Comment by jason-dai** [comment]
Also need to update APIGuide/FeatureEngineering/image.md, and list some common values for the image codec

File: pyzoo/zoo/pipeline/nnframes/nn_image_reader.py [hunk 3]
```diff
@@ -29,17 +29,21 @@ class NNImageReader:
...
+                   flags=-1, bigdl_type="float"):
         """
         Read the directory of images into DataFrame from the local or remote source.
         :param path Directory to the input data files, the path can be comma separated paths as the
                 list of inputs. Wildcards path are supported similarly to sc.binaryFiles(path).
         :param min_partitions A suggestion value of the minimal splitting number for input data.
         :param resizeH height after resize, by default is -1 which will not resize the image
         :param resizeW width after resize, by default is -1 which will not resize the image
+        :param flags specifying the color type of a loaded image, same as in OpenCV.imread.
+               By default is Imgcodecs.CV_LOAD_IMAGE_UNCHANGED(-1)
```
[the remark this answers is not in the document]
Comment (dding3) [reply pyzoo/zoo/pipeline/nnframes/nn_image_reader.py]: The default behavior is the same. In our previous implementation it's also `Imgcodecs.CV_LOAD_IMAGE_UNCHANGED`

File: pyzoo/zoo/feature/image/imageset.py (hunk 2 above)
Comment (dding3) [reply pyzoo/zoo/feature/image/imageset.py]: I found in opencv the parameter is named as `flags` and Imgcodecs is a class which has lots of method. I am wondering if we can be consistent with opencv?

Comment (jason-dai) [reply pyzoo/zoo/feature/image/imageset.py]: `flags` is somewhat confusing?

**Comment by dding3** [comment]
Updated the doc and renamed the parameter.

**Comment by jason-dai** [comment]
LGTM

**Comment by dding3** [comment]
Thanks for the review.

Jekins pass. http://10.239.47.211:18888/job/ZOO-PR-Validation/708/

**Pull request merged**